core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -531,6 +530,11 @@ private DefaultClientRequestContext(DefaultClientRequestContext ctx,
 
         this.endpointGroup = endpointGroup;
         updateEndpoint(endpoint);
+        if (ctx.endpoint() == endpoint || endpoint == null) {
+            eventLoop = ctx.eventLoop().withoutContext();
+        } else {
+            acquireEventLoop(endpoint);
+        }
```

Comment:
What happens if we just always call `acquireEventLoop()`? (maybe except when `endpoint == null`)

Reply:
It sounds like a good idea to update the `lastActivityTimeNanos`.

However if we always call `acquireEventLoop()`, the number of active requests may be increased for the initial attempt. Let me add an exception for the initial attempt.

Reply:
Also, when `ctx.endpoint() == endpoint` and the `maxNumEventLoopPerEndpoint` is greater than 1, the different eventLoop might be used which is inefficient.

Reply:
> the different eventLoop might be used which is inefficient.

Although it is inefficient I think it would be good to try a different connection than the connection that has already failed.